### ISSUE TITLE
fillTriangle() fails on 320x480 displays

### ISSUE THREAD

**Issue by prenticedavid** [adafruit/Adafruit-GFX-Library#218]
There is intermediate overflow in the "lower half" of a triangle.
It can be fixed by casting the calculation to 32-bit:

```
    // For lower part of triangle, find scanline crossings for segments
    // 0-2 and 1-2.  This loop is skipped if y1=y2.
    sa = (int32_t)dx12 * (y - y1);
    sb = (int32_t)dx02 * (y - y0);

```
David.

**Comment by ladyada** [comment]
woops - wanna try submitting a pr and @makermelissa can test? :)

**Comment by makermelissa** [comment]
Fixed by #219
